Dockerfile

```diff
@@ -1,28 +1,74 @@
 # Docker Container that runs the Nebra Diagnostics Tool
 
-FROM balenalib/raspberry-pi-debian-python:buster-run-20211014
+####################################################################################################
+################################## Stage: builder ##################################################
 
+# The balenalib/raspberry-pi-debian-python image was tested but missed many dependencies.
+FROM balenalib/raspberry-pi-debian:buster-build-20211014 as builder
+
+# Nebra uses /opt by convention
 WORKDIR /opt/
 
+# Copy python dependencies for `pip install` later
+COPY requirements.txt requirements.txt
+
+# This will be the path that venv uses for installation below
+ENV PATH="/opt/venv/bin:$PATH"
+
+# Install python3-minimal, pip3, wget, venv.
+# Then set venv environment copied from builder.
+# Finally, use pip to install dependencies.
+RUN \
+    install_packages \
+            python3-dev=3.7.3-1 \
```

Comment:
IIRC @vpetersson thought it best to unpin packages previously

Reply:
Make sense